kahuna/public/js/components/gr-delete-image/gr-delete-image.js

```diff
@@ -0,0 +1,40 @@
+import angular from 'angular';
+import '../gr-confirm-delete/gr-confirm-delete';
+
+export const imageDelete = angular.module('gr.imageDelete', ['gr.confirmDelete']);
+
+imageDelete.controller('grImageDeleteCtrl', ['$scope', 'mediaApi', function ($scope, mediaApi) {
+    var ctrl = this;
+
+    ctrl.delete = function () {
+        mediaApi.delete(ctrl.image)
+            .then((resp) => {
+                if (angular.isDefined(ctrl.onSuccess)) {
+                    ctrl.onSuccess(resp);
+                }
+            })
+            .catch((err) => {
+                if (angular.isDefined(ctrl.onError)) {
+                    ctrl.onError(err);
+                }
+            });
+    };
+}]);
+
+imageDelete.directive('grImageDelete', [function () {
```

Comment:
Is there a reason why the directive / ctrl and file name are different - cold be confusing when searching.